documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,186 @@
-*   _Message enqueuing_. The message is actually enqueued, queue is sorted by message timestamps in increasing order and counters are updated (e.g., counters for the total number of bytes in the queue).
-*   _Message drop_. In some circumstances, due to network congestion or to ongoing attacks, some messages shall be dropped to guarantee bounded delays and isolate attacker's messages. Specifically, a node shall drop messages in two situations:
-    - since buffers are of a limited size, if the total number of bytes in all queues exceeds a certain threshold, new incoming messages are dropped;
-    - to guarantee the security of the network, if a certain queue exceeds a given threshold, new incoming packets from that specific node ID will be dropped.
+* _Classification_. The mechanism identifies the queue where the message belongs to according to the node ID of
+  the message issuer.
+* _Message enqueuing_. The message is actually enqueued, queue is sorted by message timestamps in increasing order
+  and counters are updated (e.g., counters for the total number of bytes in the queue).
+* _Message skipping_. The message is skipped when they are confirmed while still waiting to be scheduled in the
+  outbox. Skipped message doesn't decrease issuing node's deficit, it is not gossipped or added to the tip pool.
+* _Message drop_. In some circumstances, due to network congestion or to ongoing attacks, some messages shall be
+  dropped to guarantee bounded delays and isolate attacker's messages. Specifically, a node shall drop messages in
+  the following situation:
+    - since the total outbox buffer is of a limited size, if the total number of bytes in all queues exceeds the
+      limit after a new message is enqueued, the oldest messages from longest mana-scaled queues are dropped until the
+      current buffer size is smaller than the maximum size. Mana scaled length of a queue is calculated by dividing
+      number of bytes in the queue by the amount of access Mana of a corresponding node:
+    - `Mana-scaled queue size = queue size / node aMana`;
 
 The dequeue mechanism includes the following components:
 
-*   _Queue selection_. A queue is selected according to round robin scheduling algorithm. In particular, we use a modified version of the deficit round robin (DRR) algorithm.
-*   _Message dequeuing_. The first message of the queue is dequeued, and list of active nodes is updated.
-*   _Scheduler management_. Scheduler counters and pointers are updated.
+* _Queue selection_. A queue is selected according to round-robin scheduling algorithm. In particular, we use a
+  modified version of the deficit round-robin (DRR) algorithm.
+* _Message dequeuing_. The first (oldest) message of the queue, that satisfies certain conditions is dequeued, and
+  list of active nodes is updated. Conditions a message must satisfy:
```

Comment:
Is the list of active nodes mentioned elsewhere? Otherwise it's not easy to understand why this is here and what's its function.